docs/book/getting-started/deploying-zenml/docker.md

```diff
@@ -1,88 +1,359 @@
-    known to work with the CLI and ZenML Dashboard. You should use the other
-    schemes only if you intent to access the REST API directly.
+If you're planning on deploying a custom containerized ZenML server yourself,
+you probably need to configure some settings for it like the database it should
+use, the default user details and more. The ZenML server container image uses
+sensible defaults, so you can simply start a container without worrying too much
+about the configuration. However, if you're looking to connect the ZenML server
+to an external MySQL database, or to persist the internal SQLite database, or
+simply want to control other settings like the default account, you can do so
+by customizing the container's environment variables.
+
+The following environment variables can be passed to the container:
+
+- **ZENML_DEFAULT_PROJECT_NAME**:
+    The name of the default project created by the server on first deployment,
+    during database initialization. Defaults to `default`.
+- **ZENML_DEFAULT_USER_NAME**:
+    The name of the default admin user account created by the server on first
+    deployment, during database initialization. Defaults to `default`.
+- **ZENML_DEFAULT_USER_PASSWORD**:
+    The password to use for the default admin user account. Defaults to an empty
+    password value, if not set.
+- **ZENML_STORE_URL**:
+    This URL should point to a SQLite database file _mounted in the container_,
+    or to a MySQL compatible database service _reachable from the container_. It
+    takes one of the forms:
+
+        sqlite:////path/to/zenml.db
     
-- ZENML_DEFAULT_PROJECT_NAME
-    The name of the default project created by the server on first deployment.
-- ZENML_DEFAULT_USER_NAME
-    The name of the default user account created by the server on first
-    deployment.
-- ZENML_DEFAULT_USER_PASSWORD
-    The password to use for the default user account.    
-
-- ZENML_STORE_TYPE
-    The type of backend store that you want to configure 
-    ZenML with. This should always be set to "sql".
-- ZENML_STORE_URL
-    This URL should point to a MysSQL compatible database. It takes the form:
+    or:
 
         mysql://username:password@host:port/database
 
-- ZENML_STORE_SSL_CA
-    The server certificate in use by the MySQL database can be passed using this
-    variable, if you are using SSL to connect to the database.
-- ZENML_STORE_SSL_CERT
-    The client certificate required to connect to the MySQL database, if you
-    are using SSL client certificates to connect to the database.
-- ZENML_STORE_SSL_KEY
-    The client certificate private key required to connect to the MySQL
-    database, if you are using SSL client certificates to connect to the
-    database.
-- ZENML_STORE_SSL_VERIFY_SERVER_CERT
-    Whether to verify SSL certificate for the MySQL server.    
+{% hint style="warning" %}
+The SQLite database backend option currently has one major limitation:
```

Comment:
Why would that be? If you mount the DB, shouldn't it be available both in the container and outside where the pipelines are run? Which information will be missing exactly?

Reply:
Even if you mount the SQLite DB in the container, it's too complicated to get the client and local (Docker) orchestrators to all access (mount and access in the case of local Docker orchestrators) the same file:

- the client and server need to use the exact same path
- the SQLite database file must reside under the `local_stores` folder, which is the only folder mounted in the docker orchestrator containers
- other complications that I can't think of right now

I prefer to discourage users from using this DB type rather than to explain the intricacies of manually managing what ZenML does automatically.